File: frontend/src/app/core/services/reminder.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { Reminder } from '../models/reminder.models';

@Injectable({
    providedIn: 'root'
})
export class ReminderService {
    private apiUrl = environment.apiBaseUrl + '/reminders';

    constructor(private http: HttpClient) { }

    getAllReminders(): Observable<Reminder[]> {
        return this.http.get<Reminder[]>(`${this.apiUrl}/`);
    }

    createReminder(dto: Partial<Reminder>): Observable<Reminder> {
        return this.http.post<Reminder>(`${this.apiUrl}/`, dto);
    }

    getOverdueReminders(): Observable<Reminder[]> {
        return this.http.get<Reminder[]>(`${this.apiUrl}/overdue/`);
    }

    markComplete(id: string): Observable<void> {
        return this.http.put<void>(`${this.apiUrl}/${id}/complete/`, {});
    }

    deleteReminder(id: string): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}/`);
    }
}

File: backend/src/main/java/com/ajay/career/service/ReminderService.java
package com.ajay.career.service;

import com.ajay.career.dto.ReminderDTO;

import java.util.List;
import java.util.UUID;

public interface ReminderService {
    ReminderDTO createReminder(UUID jobId, ReminderDTO dto);

    List<ReminderDTO> getOverdueReminders();

    List<ReminderDTO> getAllReminders();

    ReminderDTO markComplete(UUID id);
}
